scripts/SetupVulkan.py

```diff
@@ -0,0 +1,77 @@
+import os
+import sys
+import subprocess
+from pathlib import Path
+
+import Utils
+
+from io import BytesIO
+from urllib.request import urlopen
+
+class VulkanConfiguration:
+    requiredVulkanVersion = "1.2.170.0"
+    vulkanDirectory = "./Hazel/vendor/VulkanSDK"
+
+    @classmethod
+    def Validate(cls):
+        if (not cls.CheckVulkanSDK()):
+            print("Vulkan SDK not installed correctly.")
+            return
+            
+        if (not cls.CheckVulkanSDKDebugLibs()):
+            print("Vulkan SDK debug libs not found.")
+
+    @classmethod
+    def CheckVulkanSDK(cls):
+        vulkanSDK = os.environ.get("VULKAN_SDK")
+        if (vulkanSDK is None):
+            print("\nYou don't have the Vulkan SDK installed!")
+            cls.__InstallVulkanSDK()
+            return False
+        else:
+            print(f"\nLocated Vulkan SDK at {vulkanSDK}")
+
+        if (cls.requiredVulkanVersion not in vulkanSDK):
+            print(f"You don't have the correct Vulkan SDK version! (Engine requires {cls.requiredVulkanVersion})")
+            cls.__InstallVulkanSDK()
+            return False
+    
+        print(f"Correct Vulkan SDK located at {vulkanSDK}")
+        return True
+
+    @classmethod
+    def __InstallVulkanSDK(cls):
+        permissionGranted = False
+        while not permissionGranted:
+            reply = str(input("Would you like to install VulkanSDK {0:s}? [Y/N]: ".format(cls.requiredVulkanVersion))).lower().strip()[:1]
+            if reply == 'n':
+                return
+            permissionGranted = (reply == 'y')
+
+        vulkanInstallURL = f"https://sdk.lunarg.com/sdk/download/{cls.requiredVulkanVersion}/windows/VulkanSDK-{cls.requiredVulkanVersion}-Installer.exe"
+        vulkanPath = f"{cls.vulkanDirectory}/VulkanSDK-{cls.requiredVulkanVersion}-Installer.exe"
+        print("Downloading {0:s} to {1:s}".format(vulkanInstallURL, vulkanPath))
+        Utils.DownloadFile(vulkanInstallURL, vulkanPath)
+        print("Running Vulkan SDK installer...")
+        os.startfile(os.path.abspath(vulkanPath))
+        print("Re-run this script after installation!")
+        quit()
+
+    @classmethod
+    def CheckVulkanSDKDebugLibs(cls):
+        shadercdLib = Path(f"{cls.vulkanDirectory}/Lib/shaderc_sharedd.lib")
+        if True or (not shadercdLib.exists()):
```

Comment:
Some minor debug things were still in place
```suggestion
        if not shadercdLib.exists():
```